src/main/java/app/persistence/OrderMapper.java

```diff
@@ -92,8 +90,56 @@ public List<Order> getAllCustomerOrders(int customerId, ConnectionPool connectio
         return null;
     }
 
-    public void createOrder(Order order, ConnectionPool connectionPool) {
+    public static void createOrder(Order order, int customerId, ConnectionPool connectionPool) throws DatabaseException {
+        String sql = "INSERT INTO orders (account_id, order_title, carport_width, carport_length, order_status, order_total_price, order_timestamp) VALUES (?,?,?,?,?,?,?) RETURNING order_id;";
+
+        try (
+                Connection connection = connectionPool.getConnection();
+                PreparedStatement ps = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+        ) {
+            ps.setInt(1, customerId);
+            ps.setString(2, order.getTitle());
+            ps.setInt(3, order.getCarportWidth());
+            ps.setInt(4, order.getCarportLength());
+            ps.setString(5, order.getStatus().toString());
+            ps.setInt(6, order.getTotalPrice());  //tjek at totalprisen er sat til at kunne være null
+            ps.setTimestamp(7, Timestamp.valueOf(order.getTimestamp()));
+
+            int affectedRows = ps.executeUpdate();
+            if (affectedRows == 0) {
+                throw new DatabaseException("failed to create order");
+            }
 
+            try (ResultSet generatedKeys = ps.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    order.setOrderId(generatedKeys.getInt(1));
+                    createOrderBill(connectionPool, order.getOrderId(), order.getOrderBill());
+                } else {
+                    throw new DatabaseException("failed to create order");
+                }
+            }
+        } catch (SQLException e) {
+            throw new DatabaseException(e.getMessage());
+        }
+    }
+
+    private static void createOrderBill(ConnectionPool connectionPool, int orderId, List<OrderBillItem> orderBillItems) throws DatabaseException {
+        String sql = "INSERT INTO order_bills (order_id, material_variant_id, item_description, item_quantity) VALUES (?, ?, ?, ?);";
```

Comment:
There are 4 columns but there are 5 questionmarks. So one of the questionmarks should probably be deleted

Reply:
there's only 4